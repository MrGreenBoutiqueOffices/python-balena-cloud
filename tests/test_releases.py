"""Test the releases model for Balena Cloud."""

# pylint: disable=too-many-arguments,too-many-positional-arguments
from __future__ import annotations

from typing import TYPE_CHECKING

from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from balena_cloud import BalenaCloud


async def test_get_release(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    balena_cloud_client: BalenaCloud,
) -> None:
    """Test the get_release method."""
    aresponses.add(
        "api.balena-cloud.com",
        "/v7/release(1)",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("releases/release.json"),
        ),
    )
    release = await balena_cloud_client.get_release(release_id=1)
    assert release == snapshot


async def test_remove_release(
    aresponses: ResponsesMockServer,
    balena_cloud_client: BalenaCloud,
) -> None:
    """Test the remove_release method."""
    aresponses.add(
        "api.balena-cloud.com",
        "/v7/release(1)",
        "DELETE",
    )
    await balena_cloud_client.remove_release(release_id=1)
