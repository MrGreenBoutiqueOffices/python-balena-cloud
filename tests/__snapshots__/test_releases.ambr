# serializer version: 1
# name: test_get_release
  Release(id=987654, status='success', semver='0.1.0', semver_prerelease='', revision=0, created_at=datetime.datetime(2025, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_final=True, is_invalidated=False, is_passing_tests=True)
# ---
